pyresample/area_config.py

```diff
@@ -849,3 +850,37 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
     with open(yaml_area_file, 'w') as yaml_file:
         for area in areas:
             yaml_file.write(area.create_areas_def())
+
+
+def generate_area_def_rst_list(area_file):
+    """Create rst list of available area definitions with overview plot.
+
+    Args:
+        area_file (str): Path to area yaml file.
+
+    Returns:
+        str: rst list formatted string.
+    """
+    area_list = []
+
+    template = ("{area_name}\n"
+                "{n:^>{header_title_length}}\n\n"
+                ".. raw:: html\n\n"
+                "{content}\n\n"
+                "     <hr>\n\n")
+
+    for i, (aname, params) in enumerate(_read_yaml_area_file_content(area_file).items()):
+        area = _create_area_def_from_dict(aname, params)
+        if not hasattr(area, "_repr_html_"):
+            continue
+
+        if i == 0:
+            area_rep = area_repr(area, include_header=False)
+        else:
+            area_rep = area_repr(area, include_header=False, include_static_files=False)
```

Comment:
Not tested, but I think this makes sense and removes a lot of code:

```suggestion
    for aname, params in _read_yaml_area_file_content(area_file).items():
        area = _create_area_def_from_dict(aname, params)
        if not hasattr(area, "_repr_html_"):
            continue

        area_rep = area_repr(area, include_header=False, include_static_files=not bool(area_list))
```

Reply:
I will try that.

Reply:
I think if this can be added/changed, then I'm good with the rest of the PR given there is refactoring planned.

Reply:
I changed it and this works.